x-pack/plugins/security_solution/server/endpoint/services/actions/action_list.ts

```diff
@@ -274,41 +271,11 @@ const getActionDetailsList = async ({
         : categorizedResponse.item.data.action_id === action.id
     );
 
-    // find the specific response's details using that set of matching responses
-    const { isCompleted, completedAt, wasSuccessful, errors, agentState, outputs } =
-      getActionCompletionInfo(action, matchedResponses);
-
-    const { isExpired, status } = getActionStatus({
-      expirationDate: action.expiration,
-      isCompleted,
-      wasSuccessful,
-    });
+    const actionRecord = createActionDetailsRecord(action, matchedResponses, agentsHostInfo);
 
-    const actionRecord: ActionListApiResponse['data'][number] = {
-      id: action.id,
-      agents: action.agents,
-      hosts: action.agents.reduce<ActionDetails['hosts']>((acc, id) => {
-        acc[id] = { name: agentsHostInfo[id] ?? '' };
-        return acc;
-      }, {}),
-      command: action.command,
-      startedAt: action.createdAt,
-      isCompleted,
-      completedAt,
-      wasSuccessful,
-      errors: action.error?.message ? [action.error.message] : errors,
-      agentState,
-      isExpired,
-      status,
-      // 8.8 onwards, show outputs only for actions with matching requested action ids
-      outputs: withOutputs && withOutputs.includes(action.id) ? outputs : undefined,
-      createdBy: action.createdBy,
-      comment: action.comment,
-      parameters: action.parameters,
-      alertIds: action.alertIds,
-      ruleId: action.ruleId,
-      ruleName: action.ruleName,
-    };
+    if (withOutputs && !withOutputs.includes(action.id)) {
```

Comment:
I think the end result is the same here. 

If the request did not ask for outputs to be included, then we remove it from the `actionRecord` so that its not returned...

In the prior implementation, it was building the actual `ActionDetails` record and thus in that case it was adding the `outputs` only if the action id appeared in `withOutputs`.  The property `outputs` is marked as optional in the type, so `delete`'ing it is the same as assigning `undefined` in this case.

let me know if you have any other questions and thanks for the review :)